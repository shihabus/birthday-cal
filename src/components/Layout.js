import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Wrapper = styled.div`
  height: 100%;
  padding: 5vh;
  background: #eaf0f2;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 5vh 40vh calc(100% - (45vh + 4vh));
  grid-row-gap: 2vh;
  grid-template-areas: "header header" "list list" "viewer search";
`;

export default function Layout(props) {
  return <Wrapper>{props.children}</Wrapper>;
}

Layout.propTypes = {
  children: PropTypes.arrayOf(PropTypes.node)
};
