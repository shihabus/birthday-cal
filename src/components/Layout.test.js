import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Layout from "./Layout";

test("Layout snapshot", () => {
  const { container } = render(<Layout />);
  expect(container.firstChild).toMatchSnapshot();
});
