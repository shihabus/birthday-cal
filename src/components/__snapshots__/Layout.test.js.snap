// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout snapshot 1`] = `
<div
  class="sc-AxjAm etyYFM"
/>
`;
